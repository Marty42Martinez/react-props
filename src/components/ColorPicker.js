import React, { PureComponent } from 'react';
import PropTypes from 'prop-types';

export default class ColorPicker extends PureComponent {
  static propTypes = {
    title: PropTypes.string.isRequired
  }
  // eslint-disable-next-line
  clickHandler = event => {
    console.log(`${this.props.title}`);
  }

  render() {
    return (
      <button onClick={this.clickHandler}>{this.props.title}</button>
    );
  }
}
