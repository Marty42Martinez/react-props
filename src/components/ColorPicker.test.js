import React from 'react';
import { shallow } from 'enzyme';
import ColorPicker from './ColorPicker';

describe('ColorPicker component', () => {
  it('renders a ColorPicker Button', () => {
    const wrapper = shallow(<ColorPicker />);
    expect(wrapper).toMatchSnapshot();
  });
});
