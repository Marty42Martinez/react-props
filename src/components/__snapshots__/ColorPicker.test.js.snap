// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ColorPicker component renders a ColorPicker Button 1`] = `
<button
  onClick={[Function]}
/>
`;
